crates/core/src/kernel/snapshot/replay.rs

```diff
@@ -311,7 +312,7 @@ fn parse_partitions(batch: RecordBatch, partition_schema: &StructType) -> DeltaR
 
     insert_field(
         batch,
-        StructArray::try_new(
+        StructArray::try_new_with_length(
```

Comment:
- This and similar changes is the fix for https://github.com/apache/arrow-rs/pull/7539